src/cpu/Dasm.cc

```diff
@@ -108,18 +115,50 @@ unsigned dasm(const MSXCPUInterface& interface, uint16_t pc, std::span<uint8_t,
 			break;
 		}
 	}
+	// used by unittest only
 	return i;
 }
 
-unsigned instructionLength(const MSXCPUInterface& interface, uint16_t pc,
-                           EmuTime::param time)
+std::span<uint8_t> fetchInstruction(const MSXCPUInterface& interface, uint16_t addr,
+                                    std::span<uint8_t, 4> buffer, EmuTime::param time)
+{
+	uint16_t idx = 0;
+	buffer[idx++] = interface.peekMem(addr, time);
+	auto len = instr_len_tab[buffer[0]];
+	if (len >= 4) {
+		buffer[idx++] = interface.peekMem(addr + 1, time);
+		len = instr_len_tab[64 * len + buffer[1]];
+	}
+	while (idx < len) {
+		buffer[idx++] = interface.peekMem(addr + idx, time);
+	}
+	if (len == 4 && !std::binary_search(ddcb__.begin(), ddcb__.end(), buffer[3])) len = 1;
```

Comment:
Remove this line again. We don't want this "recover" behavior. But also it wouldn't work:
* Suppose we do set `len = 1`.
* Then in the next line we only return a `span` of length = 1.
* We pass this to `dasm()`.
* But the implementation of `dasm()` _will_ read `bin[1]`, and that's undefined behavior. E.g. when compiled with `-D_GLIBCXX_DEBUG` it will trigger an assert.